src/main/java/com/faforever/client/mod/ModService.java

```
@@ -308,17 +313,26 @@ private String readPreferencesFile(Path preferencesFile) throws IOException {
     Map<String, Charset> availableCharsets = Charset.availableCharsets();
     String preferencesContent = null;
     for (Charset charset : availableCharsets.values()) {
+      CharsetDecoder decoder = charset.newDecoder();
+      decoder.onMalformedInput(CodingErrorAction.REPORT);
+      decoder.onUnmappableCharacter(CodingErrorAction.REPORT);
+      log.info("Trying to read preferences file with charset: " + charset.displayName());
       try {
-        log.debug("Trying to read preferences file with charset: " + charset.displayName());
-        preferencesContent = Files.readString(preferencesFile, charset);
-        log.debug("Successfully read preferences file with charset: " + charset.displayName());
+        ByteBuffer buffer = ByteBuffer.wrap(Files.readAllBytes(preferencesFile));
+        CharBuffer charBuffer = decoder.decode(buffer);
+        preferencesContent = charBuffer.toString();
+        log.info("Successfully read preferences file with charset: " + charset.displayName());
         break;
-      } catch (Exception e) {
-        log.warn("Could not read preferences file with charset: " + charset.displayName());
+      } catch (CharacterCodingException e) {
+        log.info("Failed to read preferences file with charset: " + charset.displayName());
+        // Continue and try a different character set
+      } catch (IOException e) {
+        // User has a malformed file
+        throw new RuntimeException(e);
```

I would still somewhere just catch all exceptions just in case